dali/operators/reader/video_reader_op.h

```diff
@@ -65,20 +72,32 @@ class VideoReader : public DataReader<GPUBackend, SequenceWrapper> {
 
     DALI_ENFORCE(dtype_ == DALI_FLOAT || dtype_ == DALI_UINT8, "Data type must be FLOAT or UINT8.");
 
-    enable_label_output_ = !file_root_.empty() || !file_list_.empty();
-    DALI_ENFORCE(enable_label_output_ || !enable_frame_num_,
+    can_use_frames_timestamps_ = !file_list_.empty() || (!filenames_.empty() && has_labels_arg);
+
+    DALI_ENFORCE(can_use_frames_timestamps_ || !enable_frame_num_,
                  "frame numbers can be enabled only when "
-                 "`file_list` or `file_root` argument is passed");
-    DALI_ENFORCE(enable_label_output_ || !enable_timestamps_,
+                 "`file_list` or `filenames` argument is passed");
+    DALI_ENFORCE(can_use_frames_timestamps_ || !enable_timestamps_,
                  "timestamps can be enabled only when "
-                 "`file_list` or `file_root` argument is passed");
+                 "`file_list` or `filenames` argument is passed");
+
+    DALI_ENFORCE(!(has_labels_arg && filenames_.empty()),
+                 "The argument ``labels`` is valid only when file paths "
+                 "are provided as ``filenamess`` argument.");
+
+    if (!filenames_.empty() && has_labels_arg) {
+      DALI_ENFORCE(filenames_.size() == labels_.size() || labels_.empty(), make_string("Provided ",
+                  labels_.size(), " labels for ", filenames_.size(), " files."));
+    }
+
+    output_labels_ = has_labels_arg || !file_list_.empty() || !file_root_.empty();
```

Comment:
I'd leave it as ```output_labels_ = has_labels_arg``` and use it in line 35

Reply:
I cannot use `output_labels_ ` in L35 as `VideoReaderOutputFn` is a free funtion, and we want to output lables not only when `labels` argument is provided. This line is what it suppose to be.